src/compiler/checker.ts

```diff
@@ -929,7 +929,7 @@ module ts {
             // The ES6 spec permits export * declarations in a module to circularly reference the module itself. For example,
             // module 'a' can 'export * from "b"' and 'b' can 'export * from "a"' without error.
             function visit(symbol: Symbol) {
-                if (symbol.flags & SymbolFlags.HasExports && !contains(visitedSymbols, symbol)) {
+                if (symbol && symbol.flags & SymbolFlags.HasExports && !contains(visitedSymbols, symbol)) {

```

Comment:
we later on call `visit(resolveExternalModuleName(node, (<ExportDeclaration>node).moduleSpecifier));` and that can be undefined if the module is empty. see my second test case.